### PR TITLE
fix(projects): Auto-set employee and company in Timesheet

### PR DESCRIPTION
**Pull request by barredterra**
#### Use-case
User creates a new Timesheet.

#### Old behavior
At the moment, when creating a Timesheet, the user has to enter his name and company manually.

#### New behavior
If there is an `Employee` with matching `user_id` for the session user, set `employee` and `company` automatically.

If another person creates the Timesheet, she would have to click on the employee field and set the specific employee manually. Setting the employee will only be necessary in this special case.

screencast

Also added semicolons, where missing.

### CODE REVIEW

File: erpnext/projects/doctype/timesheet/timesheet.js [hunk 1]
```diff
@@ -20,33 +20,37 @@ frappe.ui.form.on("Timesheet", {
...
+			};
+		};
 	},
 
 	onload: function(frm){
 		if (frm.doc.__islocal && frm.doc.time_logs) {
 			calculate_time_and_amount(frm);
 		}
+
+		if (frm.doc.docstatus < 1) {
```
Comment (nabinhait) [inline erpnext/projects/doctype/timesheet/timesheet.js]: It should only set employee and company if document is unsaved.

`if(frm.is_new())`

**Review by nabinhait — DISMISSED** [review]

**Comment by barredterra** [comment]
Thanks for the hint, done. @nabinhait

**Comment by barredterra** [comment]
Anything else I can do to get this merged, @nabinhait?

**Comment by barredterra** [comment]
@sagarvora did anything happen to Nabin?

**Review by sagarvora — DISMISSED** [review]

File: erpnext/projects/doctype/timesheet/timesheet.js [hunk 2]
```diff
@@ -294,4 +298,17 @@ var calculate_time_and_amount = function(frm) {
...
+};
+
+// set employee (and company) to the one that's currently logged in
+const set_employee_and_company = function(frm) {
+	const options = { user_id: frappe.session.user };
+	const fields = ['name', 'company'];
+	frappe.db.get_value('Employee', options, fields).then(({ message }) => {
+		if (message) {
+			// there is an employee with the currently logged in user_id
+			frm.fields_dict['employee'].set_value(message.name);
```
Comment (sagarvora) [inline erpnext/projects/doctype/timesheet/timesheet.js]: Let's use `frm.set_value` for consistency instead?
https://github.com/frappe/erpnext/pull/16743/files#diff-bb8ecad0277db241f5eef7705751b0ceR300

Comment (barredterra) [reply erpnext/projects/doctype/timesheet/timesheet.js]: Okay, but only if you promise to get this merged within a week :stuck_out_tongue_winking_eye:

**Review by sagarvora — APPROVED** [review]

**Pull request merged**